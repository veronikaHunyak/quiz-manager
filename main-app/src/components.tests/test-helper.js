export const mockQuiz = [
  {
    _id: "5eb14db7c8248d0f7cd4a279",
    quizKey: "quiz0",
    quizName: "First quiz test name",
    quizData: [
      {
        question: "Question 1",
        options: [{ A: "one" }, { B: "two" }, { C: "three" }],
        answer: "B",
      },
      {
        question: "question2",
        options: [{ A: "a" }, { B: "b" }, { C: "c" }, { D: "d" }],
        answer: "C",
      },
    ],
  },
];
