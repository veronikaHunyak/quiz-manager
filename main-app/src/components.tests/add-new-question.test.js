import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import NewQuestion from "../components/add-new-question";
import { mockQuiz } from "./test-helper";

describe("NewQuestion", () => {
  const props = {
    newQuestion: false,
    quiz: {},
    onQuizUpdated: jest.fn(),
  };

  it("should render correctly", () => {
    expect(toJson(shallow(<NewQuestion {...props} />))).toMatchSnapshot();
  });

  it("should match snapshot when there is a quiz", () => {
    const wrapper = shallow(<NewQuestion {...props} quiz={mockQuiz[0]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should match snapshot when newQuestion is true", () => {
    const wrapper = shallow(<NewQuestion {...props} newQuestion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
