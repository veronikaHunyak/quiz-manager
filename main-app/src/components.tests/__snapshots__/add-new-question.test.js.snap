// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion should match snapshot when newQuestion is true 1`] = `
<div>
  <div />
  <div>
    <button
      class="btn btn-danger"
      onClick={[Function]}
      style={
        Object {
          "float": "right",
          "margin": "10px",
        }
      }
      type="button"
    >
      <Check
        color="currentColor"
        size="1em"
        style={
          Object {
            "marginRight": "10px",
          }
        }
      />
      Add Question(s)
    </button>
  </div>
</div>
`;

exports[`NewQuestion should match snapshot when there is a quiz 1`] = `
<div>
  <div>
    <div
      key="0"
      style={
        Object {
          "margin": "35px",
        }
      }
    >
      <div
        class="row"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <div
          class="col-2"
        >
          <label
            for="email"
          >
            Question 
            1
            :
          </label>
        </div>
        <div
          class="col-8"
        >
          <input
            class="form-control"
            id="question0"
            onChange={[Function]}
            placeholder="Enter question"
            type="text"
          />
        </div>
        <div
          class="col-2"
        >
          <button
            class="btn btn-outline-danger"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "lavenderblush",
              }
            }
            type="button"
          >
            <PlusSquare
              color="currentColor"
              size="1em"
              style={
                Object {
                  "marginRight": "10px",
                }
              }
            />
            Add answers
          </button>
        </div>
      </div>
      <NewAnswer
        questionIndex={0}
        quiz={
          Object {
            "_id": "5eb14db7c8248d0f7cd4a279",
            "quizData": Array [
              Object {
                "answer": "B",
                "options": Array [
                  Object {
                    "A": "one",
                  },
                  Object {
                    "B": "two",
                  },
                  Object {
                    "C": "three",
                  },
                ],
                "question": "Question 1",
              },
              Object {
                "answer": "C",
                "options": Array [
                  Object {
                    "A": "a",
                  },
                  Object {
                    "B": "b",
                  },
                  Object {
                    "C": "c",
                  },
                  Object {
                    "D": "d",
                  },
                ],
                "question": "question2",
              },
            ],
            "quizKey": "quiz0",
            "quizName": "First quiz test name",
          }
        }
      />
    </div>
    <div
      key="1"
      style={
        Object {
          "margin": "35px",
        }
      }
    >
      <div
        class="row"
        style={
          Object {
            "margin": "10px",
          }
        }
      >
        <div
          class="col-2"
        >
          <label
            for="email"
          >
            Question 
            2
            :
          </label>
        </div>
        <div
          class="col-8"
        >
          <input
            class="form-control"
            id="question1"
            onChange={[Function]}
            placeholder="Enter question"
            type="text"
          />
        </div>
        <div
          class="col-2"
        >
          <button
            class="btn btn-outline-danger"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "lavenderblush",
              }
            }
            type="button"
          >
            <PlusSquare
              color="currentColor"
              size="1em"
              style={
                Object {
                  "marginRight": "10px",
                }
              }
            />
            Add answers
          </button>
        </div>
      </div>
      <NewAnswer
        questionIndex={1}
        quiz={
          Object {
            "_id": "5eb14db7c8248d0f7cd4a279",
            "quizData": Array [
              Object {
                "answer": "B",
                "options": Array [
                  Object {
                    "A": "one",
                  },
                  Object {
                    "B": "two",
                  },
                  Object {
                    "C": "three",
                  },
                ],
                "question": "Question 1",
              },
              Object {
                "answer": "C",
                "options": Array [
                  Object {
                    "A": "a",
                  },
                  Object {
                    "B": "b",
                  },
                  Object {
                    "C": "c",
                  },
                  Object {
                    "D": "d",
                  },
                ],
                "question": "question2",
              },
            ],
            "quizKey": "quiz0",
            "quizName": "First quiz test name",
          }
        }
      />
    </div>
  </div>
  <div>
    <button
      class="btn btn-danger"
      onClick={[Function]}
      style={
        Object {
          "float": "right",
          "margin": "10px",
        }
      }
      type="button"
    >
      <Check
        color="currentColor"
        size="1em"
        style={
          Object {
            "marginRight": "10px",
          }
        }
      />
      Add Question(s)
    </button>
  </div>
</div>
`;

exports[`NewQuestion should render correctly 1`] = `
<div>
  <div />
  <div>
    <button
      class="btn btn-danger"
      onClick={[Function]}
      style={
        Object {
          "float": "right",
          "margin": "10px",
        }
      }
      type="button"
    >
      <Check
        color="currentColor"
        size="1em"
        style={
          Object {
            "marginRight": "10px",
          }
        }
      />
      Add Question(s)
    </button>
  </div>
</div>
`;
